### PR TITLE
Have PR release use docker images

### PR DESCRIPTION
**Pull request by LTA-Thinking**
## Description
Change the PR release to use a docker image instead of self compiled source.

## Testing
Manual testing

### CODE REVIEW

File: build/jobs/provision-deploy.yml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+     
+        $additionalProperties = $flattenedCors
+     
+        $additionalProperties["SqlServer__DeleteAllDataOnStartup"] = "true"
+        $additionalProperties["ASPNETCORE_FORWARDEDHEADERS_ENABLED"] = "true"
+
+        $templateParameters = @{
+            fhirVersion = "${{ parameters.version }}"
+            # appServicePlanResourceGroup = "${{ parameters.appServicePlanResourceGroup }}"
+            # appServicePlanName = "${{ parameters.appServicePlanName }}"
```
Comment (brendankowitz) [inline build/jobs/provision-deploy.yml:56]: Not needed now?

File: build/jobs/run-tests.yml [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+      artifactName: 'IntegrationTests'
+
+  - task: UseDotNet@2
+    inputs:
+      useGlobalJson: true
+
+  - task: AzureKeyVault@1
+    displayName: 'Azure Key Vault: ${{ parameters.keyVaultName }}'
+    inputs:
+      azureSubscription: 'Microsoft Health Open Source Subscription'
```
Comment (brendankowitz) [inline build/jobs/run-tests.yml:33]: $(ConnectedServiceName)

Comment (LTA-Thinking) [reply build/jobs/run-tests.yml:33]: I'll fix this in the yaml PR. It'll show up here when I merge from master.

File: build/jobs/provision-deploy.yml (hunk 1 above)
Comment (LTA-Thinking) [reply build/jobs/provision-deploy.yml:56]: Not needed currently. The shared App Service Plans don't work with the linux containers, but I think we will still want shared App Service Plans for cost reasons. I'll make new shared ones.

Comment (LTA-Thinking) [reply build/jobs/provision-deploy.yml]: Annoyingly I also had to make a new resource group as app service plans for windows and linux can't be in the same resource group and region. The shared plan is in msh-fhir-pr-linux.

**Pull request merged**